packages/next/src/build/swc/jest-transformer.ts

```diff
@@ -29,49 +29,79 @@ DEALINGS IN THE SOFTWARE.
 import vm from 'vm'
 import { transformSync } from './index'
 import { getJestSWCOptions } from './options'
+import type {
+  TransformerCreator,
+  TransformOptions,
+  SyncTransformer,
+} from '@jest/transform'
+import type { Config } from '@jest/types'
+import type { NextConfig, ExperimentalConfig } from '../../server/config-shared'
+
+export interface JestTransformerConfig {
+  jsConfig: any
+  resolvedBaseUrl?: string
+  pagesDir?: string
+  hasServerComponents?: boolean
+  isEsmProject: boolean
+  modularizeImports?: NextConfig['modularizeImports']
+  swcPlugins: ExperimentalConfig['swcPlugins']
+  compilerOptions: NextConfig['compiler']
+}
 
 // Jest use the `vm` [Module API](https://nodejs.org/api/vm.html#vm_class_vm_module) for ESM.
 // see https://github.com/facebook/jest/issues/9430
 const isSupportEsm = 'Module' in vm
 
-function getJestConfig(jestConfig: any) {
+function getJestConfig(
+  jestConfig: TransformOptions<JestTransformerConfig>
+): Config.ProjectConfig {
   return 'config' in jestConfig
     ? // jest 27
       jestConfig.config
     : // jest 26
-      jestConfig
+      (jestConfig as unknown as Config.ProjectConfig)
```

Comment:
The `ProjectConfig` type represents already normalized config. Configs supplied by a user are defined in types prefixed with `Initial`: `InitialConfig`, `InitialProjectConfig`.

Reply:
Jest is passing `ProjectConfig` to transformers, that is right. If transformer is called directly, you should pass `ProjectConfig` to it.

If we talk about the type of the object of the default export of `jest.config` file, this is `InitialOptions`.

If a function is exported it should return `InitialOption`.

The `projects` option if typed as: `Array<string | InitialProjectOptions>`.

Reply:
sorry, I'm not following -- this is in the transformer, this is not part of the `jest.config` file -- am I missing something?

Reply:
Must be right then.

Sorry, my mind got stuck with the problem you had with a function ending up in `jest.config`. Did not realise you are tweaking something inside a transformer.

Reply:
got it!  yeah this change ended up needing to go deeper than I expected, mostly because everything was typed as `any` which made it hard to be confident about the change until I added real types